for-users/software/ddl.rst

```diff
@@ -0,0 +1,262 @@
+
+* `Pytorch 1.1.0 <https://www.ibm.com/support/knowledgecenter/SS5SF7_1.6.1/navigation/wmlce_getstarted_pytorch.html>`_
+
+* `Caffe(IBM-enhanced) 1.0.0 <https://www.ibm.com/support/knowledgecenter/SS5SF7_1.6.1/navigation/wmlce_getstarted_caffe.html>`_
+
+* `Horovod (IBM-DDL Backend) <https://github.com/horovod/horovod>`_
+
+For a complete list of packages and their versions please see: `WMLC CE Software Packages <https://www.ibm.com/support/knowledgecenter/SS5SF7_1.6.1/navigation/wmlce_software_pkgs.html>`_.
+
+Running DDL Jobs
+================
+
+IBM DDL provides a tool called ``ddlrun`` which facilitates the launching of
+DDL jobs on Summit. When called inside of a ``bsub`` script, ``ddlrun`` will
+automatically distribute the training job across all the compute hosts in the
+reservation. ``ddlrun`` automatically selects default arguments optimized
+for performance.
+
+Basic DDL BSUB Script
+---------------------
+
+The following bsub script will run a distributed Tensorflow resnet50 training job
+across 2 nodes.
+
+.. code-block:: bash
+    :caption: script.bash
+
+    #BSUB -P <PROJECT>
+    #BSUB -W 0:10
+    #BSUB -nnodes 2
+    #BSUB -q batch
+    #BSUB -J ddl_test_job
+    #BSUB -o /ccs/home/<user>/job%J.out
+    #BSUB -e /ccs/home/<user>/job%J.out
+
+    module load ibm-wml-ce
+
+    ddlrun python $CONDA_PREFIX/tf_cnn_benchmarks/tf_cnn_benchmarks.py --variable_update=ddl --model=resnet50
+
+``bsub`` is used to launch the script as follows:
+
+.. code-block:: bash
+
+    bsub script.bash
+
+For more information on ``bsub`` and job submission
+please see: :ref:`running-jobs`.
+
+Troubleshooting Tips
+--------------------
+
+Full command
+^^^^^^^^^^^^
+
+The output from ``ddlrun`` includes the exact command used to launch the
+distributed job. This is useful if a user wants to see exactly what ``ddlrun``
+is doing. The following is the first line of the output from the above script:
+
+.. code-block:: console
+
+    $ module load ibm-wml-ce
+    (ibm-wml-ce-1.6.1) $ ddlrun python $CONDA_PREFIX/tf_cnn_benchmarks/tf_cnn_benchmarks.py --variable_update=ddl --model=resnet50
+    + /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/xl-16.1.1-3/spectrum-mpi-10.3.0.1-20190611-aqjt3jo53mogrrhcrd2iufr435azcaha/bin/mpirun -x LSB_JOBID -x PATH -x PYTHONPATH -x LD_LIBRARY_PATH -x LSB_MCPU_HOSTS -x NCCL_LL_THRESHOLD=0 -x NCCL_TREE_THRESHOLD=0 -disable_gdr -gpu --rankfile /tmp/DDLRUN/DDLRUN.xoObgjtixZfp/RANKFILE -x "DDL_OPTIONS=-mode p:6x2x1x1 " -n 12 -mca plm_rsh_num_concurrent 12 -x DDL_HOST_PORT=2200 -x "DDL_HOST_LIST=g28n14:0,2,4,6,8,10;g28n15:1,3,5,7,9,11" bash -c 'source /sw/summit/ibm-wml-ce/anaconda-base/etc/profile.d/conda.sh && conda activate /sw/summit/ibm-wml-ce/anaconda-base/envs/ibm-wml-ce-1.6.1-1 > /dev/null 2>&1 && python /sw/summit/ibm-wml-ce/anaconda-base/envs/ibm-wml-ce-1.6.1-1/ddl-tensorflow/examples/mnist/mnist-env.py'
```

Comment:
Is there anything that can be done here so that we don't have a giant horizontal scroll?

Reply:
yes, better. thanks